src/runtime/hexagon/hexagon/hexagon_device_api_v2.cc

```diff
@@ -163,6 +170,63 @@ TVM_REGISTER_GLOBAL("device_api.hexagon.mem_copy").set_body([](TVMArgs args, TVM
   *rv = static_cast<int32_t>(0);
 });
 
+std::map<void*, HexagonBuffer*> vtcmallocs;
+
+TVM_REGISTER_GLOBAL("device_api.hexagon.AllocNd").set_body([](TVMArgs args, TVMRetValue* rv) {
```

Comment:
Instead of using a lambda for the body, can we move this to a separate function?  It tends to make debugging easier later on, and becomes a template for a generalized function we can propose adding to `c_runtime_api.cc`.

Reply:
This PR creates a separate function called `AllocVtcmWorkspace` in the Hexagon Device API that is called from the lambda and which mirrors `AllocTextureWorkspace` in the OpenCL device API.  These two APIs are the template for a more generalized Device API to "allocate a workspace with storage scope" but this should occur in a follow-up PR, in my opinion.

Note that the lambda must construct `Device` and `DLDataType` arguments for `AllocVtcmWorkspace` from primitive types passed through the builtin before making the call to `AllocVtcmWorkspace`.